src/BenchmarkDotNet/Toolchains/ToolchainExtensions.cs

```diff
@@ -34,12 +35,17 @@ internal static IToolchain GetToolchain(this Runtime runtime, Descriptor descrip
             switch (runtime)
             {
                 case ClrRuntime clrRuntime:
-                    if (RuntimeInformation.IsNetCore || preferMsBuildToolchains)
-                        return clrRuntime.RuntimeMoniker != RuntimeMoniker.NotRecognized
-                            ? GetToolchain(clrRuntime.RuntimeMoniker)
-                            : CsProjClassicNetToolchain.From(clrRuntime.MsBuildMoniker);
-
-                    return RoslynToolchain.Instance;
+                    if (!preferMsBuildToolchains && RuntimeInformation.IsFullFramework &&
```

Comment:
the `RoslynToolchain` should also be used for classic Mono:

```suggestion
                    if (!preferMsBuildToolchains && (RuntimeInformation.IsFullFramework || RuntimeInformation.IsOldMono) &&
```

Reply:
Is that valid? I thought Mono would have to use a MonoRuntime.

Reply:
@timcassell you are right, it's not. Great catch! 👍